wagtail/core/blocks/struct_block.py

```diff
@@ -137,17 +137,52 @@ def clean(self, value):
 
         return self._to_struct_value(result)
 
-    def to_python(self, value):
-        """ Recursively call to_python on children and return as a StructValue """
-        return self._to_struct_value([
-            (
-                name,
-                (child_block.to_python(value[name]) if name in value else child_block.get_default())
-                # NB the result of get_default is NOT passed through to_python, as it's expected
-                # to be in the block's native type already
+    def bulk_to_python(self, values):
+        """ Bulk load the data for fields that implement bulk_to_python """
+        bulk_in_python = {}
+
+        # Go through each child_block and pre_load data if `bulk_to_python` is available.
+        for name, field in self.child_blocks.items():
+            if not hasattr(field, 'bulk_to_python'):
+                continue
+
+            bulk_in_python[name] = field.bulk_to_python(
+                [value[name] for value in values if name in value]
```

Comment:
This is liable to leave the element indexes out-of-step, if one of the keys is missing from a dict at the beginning or middle of the list (which could happen if a new field is added to an existing StructBlock definition).

Reply:
Should it use `None` for when no value comes? It is not possible to use `child_block.get_default()` as that is already a python object (value).